modules/core/04-channel/keeper/upgrade.go

```diff
@@ -68,3 +72,158 @@ func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID stri
 		LatestSequenceSend: seq - 1,
 	}, nil
 }
+
+// ChanUpgradeTry is called by a module to accept the first step of a channel upgrade
+// handshake initiated by a module on another chain. If this function is successful, the upgrade sequence
+// will be returned. If an error occurs in the callback, 0 will be returned but the upgrade sequence will
+// be incremented.
+func (k Keeper) ChanUpgradeTry(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	proposedUpgradeFields types.UpgradeFields,
+	proposedUpgradeTimeout types.UpgradeTimeout,
+	counterpartyProposedUpgrade types.Upgrade,
+	counterpartyUpgradeSequence uint64,
+	proofCounterpartyChannel,
+	proofUpgrade []byte,
+	proofHeight clienttypes.Height,
+) (proposedUpgrade types.Upgrade, err error) {
```

Comment:
I would be in favour of removing the named return args if possible

Reply:
i guess now it's a bit more explicit than when it was just the seq number